toolkit/offchain/src/init_governance/mod.rs

```diff
@@ -128,3 +130,38 @@ pub async fn run_init_governance<
 
 	Ok(result.transaction)
 }
+
+pub async fn get_governance_utxo<T: QueryLedgerState + Transactions + QueryNetwork>(
+	genesis_utxo: UtxoId,
+	client: &T,
+) -> anyhow::Result<OgmiosUtxo> {
+	let network = client.shelley_genesis_configuration().await?.network;
+
+	let (_, version_oracle_policy, validator_address) =
+		scripts_data::version_scripts_and_address(genesis_utxo, network.to_csl())?;
+
+	let utxos = client.query_utxos(&[validator_address]).await?;
+
+	let governance_utxo = utxos
+		.iter()
+		.find(|utxo| {
+			let correct_datum = utxo
+				.datum
+				.as_ref()
+				.and_then(|datum| {
+					PlutusData::from_bytes(datum.bytes.clone()).ok().and_then(|plutus_data| {
+						VersionOracleDatum::try_from(plutus_data)
+							.ok()
+							.map(|data| data.version_oracle == 32)
+					})
+				})
+				.unwrap_or(false);
+
+			let contains_version_oracle_token =
+				utxo.value.native_tokens.contains_key(&version_oracle_policy.script_hash());
+			correct_datum && contains_version_oracle_token
+		})
+		.ok_or_else(|| anyhow!("Could not find governance versioning UTXO"))?;
+
+	Ok(governance_utxo.clone())
```

Comment:
Is this clone() mandatory?

Reply:
`governance_utxo` was a ref returned from iter-find. changed to into-iter